src/rgw/rgw_rest_s3.cc

```diff
@@ -1713,8 +1713,26 @@ int RGWPostObj_ObjStore_S3::get_policy()
 
     op_ret = rgw_get_user_info_by_access_key(store, s3_access_key, user_info);
     if (op_ret < 0) {
+      class S3Extractor:public RGWS3V2AuthEngine::Extractor {
```

Comment:
Is the `S3Extractor` a makeshift solution? At the moment it only stores already extracted data. Is there a plan to make it also responsible for extracting them from `req_state`?